File: py/Program/ParkingLot.py
class ParkingLot:
    __nama = ""
    __luas = 0
    __listVehicle = []
    __kapasitas = 0
    __jmlVehicle = 0

    def __init__(self, nama = "", luas = 0, listVehicle = [], kapasitas = 0):
        self.__nama = nama
        self.__luas = luas
        self.__listVehicle = listVehicle
        self.__kapasitas = kapasitas
        self.__jmlVehicle = len(listVehicle)

    def setNama(self, nama):
        self.__nama = nama
    
    def getNama(self):
        return self.__nama
    
    def setLuas(self, luas):
        self.__luas = luas
    
    def getLuas(self):
        return self.__luas
    
    def addListVehicle(self, listVehicle):
        self.__listVehicle.append(listVehicle)
        self.__jmlVehicle += 1

    def setListVehicle(self, listVehicle):
        self.__listVehicle = listVehicle
        self.__jmlVehicle = len(listVehicle)
    
    def getListVehicle(self):
        return self.__listVehicle
    
    def setKapasitas(self, kapasitas):
        self.__kapasitas = kapasitas
    
    def getKapasitas(self):
        return self.__kapasitas
    
    def getJmlVehicle(self):
        return self.__jmlVehicle

File: cpp/Program/ParkingLot.cpp
#pragma once
#include <bits/stdc++.h>
#include "Vehicle.cpp"

using namespace std;

class ParkingLot{
    // atribut
    private:
        string nama;
        int luas;
        vector<Vehicle*> listVehicle;
        int kapasitas;
        int jmlVehicle;

    // method
    public:
        // constructor
        ParkingLot(){
            
        }

        ParkingLot(string nama, int luas, vector<Vehicle*> listVehicle, int kapasitas){
            this->nama = nama;
            this->luas = luas;
            this->listVehicle = listVehicle;
            this->kapasitas = kapasitas;
            this->jmlVehicle = listVehicle.size();
        }

        // setter dan getter
        void setNama(string nama){
            this->nama = nama;
        }

        string getNama(){
            return this->nama;
        }

        void setLuas(int luas){
            this->luas = luas;
        }

        int getLuas(){
            return this->luas;
        }

        void setListVehicle(vector<Vehicle*> listVehicle){
            this->listVehicle = listVehicle;
            this->jmlVehicle = listVehicle.size();
        }

        vector<Vehicle*> getListVehicle(){
            return this->listVehicle;
        }

        void setKapasitas(int kapasitas){
            this->kapasitas = kapasitas;
        }

        int getKapasitas(){
            return this->kapasitas;
        }

        int getJmlVehicle(){
            return this->jmlVehicle;
        }

        // menambah 1 data kendaraan baru
        void addListVehicle(Vehicle* vehicle){
            this->listVehicle.push_back(vehicle);
            this->jmlVehicle++;
        }

        // destructor
        ~ParkingLot(){
            
        }
};